olive/passes/onnx/model_builder.py

```diff
@@ -216,7 +216,9 @@ def _run_for_config(
         with open(genai_config_filepath) as istrm:
             genai_config = json.load(istrm)
 
-        genai_config["search"] = {**genai_config.get("search", {}), **config.get("search", {})}
+        new_search_options = config.get("search") or {}
+        new_search_options.update(genai_config.get("search") or {})
```

Comment:
I think the order the opposite. we want `genai_config.update(config.get("search) or {})`

Reply:
btw, python doesn't complain about common keys in the `{**a, **b}` solution 
![image](https://github.com/microsoft/Olive/assets/94929125/0dc6a9fa-b90c-44d0-846a-4451efc5c53a)

`genai_config["search"] = {**genai_config.get("search", {}), **(config.get("search") or {})}` also works.

Reply:
You are right. Updated like:
`genai_config["search"] = {**(genai_config.get("search") or {}), **(config.get("search") or {})}`

to ensure it is safe to merge genai_config, config's search config.